### ISSUE TITLE
Seems it has limited PSG sound's pitch?

### ISSUE THREAD

**Issue by Yave-Yu** [mgba-emu/mgba#1930]
I hear "Rocket Radio" sound and discover an odd.
Here's sample between Gambatte and mGBA, you can hear it difference clearly.
[Rocket Radio sound comparation.zip](https://github.com/mgba-emu/mgba/files/5499017/Rocket.Radio.sound.comparation.zip)
Other same thing is MMBN series' "jack in" sound.

**Comment by endrift** [comment]
Please retest in 76a8f4da2bb4fc4ef10a64eae5d3da0d7846c73f or newer.

**Comment by Yave-Yu** [comment]
Yes, in GBA part PSG audio are normal now! High pitch plays well (such as "jack in" sound)... but the PSG audio I still feel it's not very clear though, I don't know is it same as an actual GBA.

In GB/GBC part, the PSG audio still have an issue, the "Rocket Radio" sound still "eats" high pitch, and whole PSG audio is not very clear.
[Rocket Radio In 7512-0a674dd37.zip](https://github.com/mgba-emu/mgba/files/8823118/Rocket.Radio.In.7512-0a674dd37.zip)

**Comment by Yave-Yu** [comment]
I guess mGBA need a low-pass filter?
